app/main.py

```diff
@@ -1,3 +1,54 @@
-def shop_trip():
-    # write your code here
-    pass
+from datetime import datetime
+
+from app.trip_cost import calculate_trip_cost
+from app.file_opening import file_open
+
+
+def shop_trip() -> None:
+    user_data = file_open("config.json")
+    fuel_price = user_data["fuel_price"]
+    customers = user_data["customers"]
+    shops = user_data["shops"]
+    for customer in customers:
+        cheeper_shop = dict()
+        print(f"{customer.name} has {customer.money} dollars")
+        for shop in shops:
+            cost = calculate_trip_cost(customer, shop, fuel_price)
+            cheeper_shop.update({cost: shop})
+            print(f"{customer.name}'s trip "
+                  f"to the {shop.name} costs "
+                  f"{cost}")
+        min_cost = min(list(cheeper_shop.keys()))
+        if min_cost <= customer.money:
+            customer.location = cheeper_shop[min_cost].location
+            print(f"{customer.name} rides to {cheeper_shop[min_cost].name}\n")
+        else:
+            print(f"{customer.name} doesn't have enough money "
+                  f"to make a purchase in any shop")
+            return
```

Comment:
The `return` statement inside the `else` block will terminate the `shop_trip` function entirely if one customer doesn't have enough money. This means that subsequent customers will not be processed. Consider using `continue` instead of `return` to skip to the next customer.